src/pywifes/splice.py

```diff
@@ -0,0 +1,330 @@
+import astropy.io.fits as fits
+import numpy as np
+import scipy.sparse as sp
+
+
+def is_evenly_sampled(_x):
+    dx = _x[1:] - _x[:-1]
+    return abs((dx.max() - dx.min()) / np.median(dx)) < 1e-3
+
+
+def _A_lanczos(_x, x, a=2, missing=np.nan):
+    """
+    Returns transformation matrix for Lanczos (sinc) interpolation
+
+    Inputs
+       _x:  x-values at which the original function is sampled
+        x:  x-values to which we want to interpolate
+        a:  integer order of the kernel
+        missing:  value to use for missing data (default:  np.nan)
+    Outputs
+        A:  sparse representation of the transformation matrix
+    """
+    if not is_evenly_sampled(_x):
+        print("_A_lanczos:  warning -- _x not evenly sampled!")
+        print("Lanczos interpolation may not give good results.")
+    # Figure out which x values are in range
+    x = np.atleast_1d(x)
+    ok = np.all([x >= _x[0], x < _x[-1]], axis=0)
+    # Find bins and fractional positions within bins, and interpolate
+    fi = np.array((len(_x) - 1) * (x - _x[0]) / (_x[-1] - _x[0]))
+    i = np.array([int(f) for f in fi])
+    # Write down the matrix for Lanczos interpolation
+    EPSILON = np.finfo(np.double).eps
+    # The following line of code results in a deprecation waring
+    # A sparce matrix in link list format
+    A = sp.lil_matrix((len(x), len(_x)))
+    for m in np.arange(len(x)):
```

Comment:
I'd use `range` instead of `np.arange` for loops. `range` generates numbers as needed, whereas I think `np.arrange` will build an entire array of length len(x), then iterate through it.

Reply:
Done1